scripts/run-integ-tests.ps1

```diff
@@ -12,40 +12,39 @@
 # permissions and limitations under the License.
 
 Param (
-  [string]$Platform="windows2016"
+  [string]
+  [ValidateSet("windows2016","windows2019","windows20H2", "windows2022")]
+  $Platform="windows2016"
 )
 
 if ($Platform -like "windows2016") {
-  $BaseImageName="mcr.microsoft.com/windows/servercore@sha256:42be24b8810c861cc1b3fe75c5e99f75061cb45fdbae1de46d151c18cc8e6a9a"
-  $BaseImageNameWithDigest="mcr.microsoft.com/windows/servercore@sha256:42be24b8810c861cc1b3fe75c5e99f75061cb45fdbae1de46d151c18cc8e6a9a"
+  $BaseImageName="mcr.microsoft.com/windows/servercore:ltsc2016"
 } elseif ($Platform -like "windows2019")  {
-  $BaseImageName="mcr.microsoft.com/windows/servercore@sha256:cc6d6da31014dceab4daee8b5a8da4707233f4ef42eaf071e45cee044ac738f4"
-  $BaseImageNameWithDigest="mcr.microsoft.com/windows/servercore@sha256:cc6d6da31014dceab4daee8b5a8da4707233f4ef42eaf071e45cee044ac738f4"
-} elseif ($Platform -like "windows2004")  {
-  $BaseImageName="mcr.microsoft.com/windows/servercore@sha256:057f2a4da3777db6de54c41029439227537e5bf805de3d90b0cfd300ffbf3db0"
-  $BaseImageNameWithDigest="mcr.microsoft.com/windows/servercore@sha256:057f2a4da3777db6de54c41029439227537e5bf805de3d90b0cfd300ffbf3db0"
+  $BaseImageName="mcr.microsoft.com/windows/servercore:ltsc2019"
 } elseif ($Platform -like "windows20h2")  {
-  $BaseImageName="mcr.microsoft.com/windows/servercore@sha256:64ada3cbc39ee8152f45b6e2284e8eb0a9fc14edef5be0f78397f0d1a0879451"
-  $BaseImageNameWithDigest="mcr.microsoft.com/windows/servercore@sha256:64ada3cbc39ee8152f45b6e2284e8eb0a9fc14edef5be0f78397f0d1a0879451"
+  $BaseImageName="mcr.microsoft.com/windows/servercore:20H2"
 } elseif ($Platform -like "windows2022")  {
-  $BaseImageName="mcr.microsoft.com/windows/servercore@sha256:8f756a7fd4fe963cc7dd2c3ad1597327535da8e8f55a7d1932780934efa87e04"
-  $BaseImageNameWithDigest="mcr.microsoft.com/windows/servercore@sha256:8f756a7fd4fe963cc7dd2c3ad1597327535da8e8f55a7d1932780934efa87e04"
+  $BaseImageName="mcr.microsoft.com/windows/servercore:ltsc2022"
 } else {
   echo "Invalid platform parameter"
   exit 1
 }
 
-$ProgramFiles="C:\Program Files\Amazon\ECS"
-$env:BASE_IMAGE_NAME=$BaseImageName
-$env:BASE_IMAGE_NAME_WITH_DIGEST=$BaseImageNameWithDigest
-
 # Prepare windows base image
 $dockerImages = Invoke-Expression "docker images"
 if (-Not ($dockerImages -like "*$BaseImageName*")) {
   Invoke-Expression "docker pull $BaseImageName"
 }
 Invoke-Expression "docker tag $BaseImageName amazon-ecs-ftest-windows-base:make"
 
+$imgDigest = Invoke-Expression "docker image inspect --format `"{{.RepoDigests}}`" $BaseImageName"
+# The template returns an array so its [$BaseImageDigest]. Remove the '[' and ']'
+$imgDigest.SubString(1, $imgDigest.Length - 2)
```

Comment:
Should this be `$imgDigest = $imgDigest.SubString(1, $imgDigest.Length - 2)`?

Reply:
How the heck. That is saved in my local file. I'm not sure how git did this. Will update you are totally right